### PR TITLE
Add SpiDevinceInfo native layer support with reference STM32F4 implementation

### PR DESCRIPTION
**Pull request by smaillet-ms**
Fix for #143 
- updated Windows.Devices interop table
- Added SpiBusInfo Man/Max Frequency properties as internal calls to pass on to the HAL/PAL to provide hardware specific values.
- Implemented New CPU_SPI_xxx APIs for STM32F4. Currently hardcoded to standard values as there's not a lot of value in anything more sophisticated.
- removed redundant code used to eliminate warnings in favor of #pragma as code is a storage and perf overhead not worth the price to silence a bening warning.

### CODE REVIEW

File: CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiBusInfo.cpp [hunk 1]
```diff
@@ -2,15 +2,26 @@
 // Copyright (c) Microsoft Corporation.  All rights reserved.
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#include "windows_devices.h"
+#include "Windows_Devices.h"
```
Comment (ColinA-MSFT) [inline CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiBusInfo.cpp]: Original casing matches what's in the file system.  This should be reverted.

File: DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp [hunk 2]
```diff
@@ -59,7 +60,9 @@ void CPU_SPI_Uninitialize()
 BOOL CPU_SPI_nWrite16_nRead16( const SPI_CONFIGURATION& Configuration, UINT16* Write16, INT32 WriteCount, UINT16* Read16, INT32 ReadCount, INT32 ReadStartOffset )
 {
     NATIVE_PROFILE_HAL_PROCESSOR_SPI();
-    if(!CPU_SPI_Xaction_Start( Configuration )) return FALSE;
+    if(!CPU_SPI_Xaction_Start( Configuration ))
+        return FALSE;
```
Comment (ColinA-MSFT) [inline DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp]: Braces.

Comment (smaillet-ms) [inline DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp]: I disagree, the redundant braces are distracting for trivial statements and don't really contribute to readability.

File: Framework/Core/Windows/Devices/Spi/SpiDevice.cs [hunk 3]
```diff
@@ -22,38 +26,20 @@ public sealed class SpiDevice : IDisposable
         /// </summary>
         /// <param name="deviceId">The unique name of the device.</param>
         /// <param name="settings">Settings to open the device with.</param>
-        internal SpiDevice(string deviceId, SpiConnectionSettings settings)
+        internal SpiDevice( string deviceId, SpiConnectionSettings settings )
```
Comment (ColinA-MSFT) [inline Framework/Core/Windows/Devices/Spi/SpiDevice.cs]: Major reformatting changes should be split into separate commits.  Small ones are fine, but it's really hard to see where the functional changes are here.

File: DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp (hunk 2 above)
Comment (ColinA-MSFT) [inline DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp]: This is a nearly universal standard, shared by other Microsoft .NET teams (e.g. Roslyn), and in C# will generate StyleCop errors.  I realize this is C++, but we should practice the same safe engineering practices.

File: Framework/Core/Windows/Devices/Spi/SpiDevice.cs [hunk 4]
```diff
@@ -248,24 +234,41 @@ internal static string[] GetValidBusNames()
...
+            {
+                if( spiBusNames[ i ] == deviceId )
+                    retVal = i;
+            }
 
-        [MethodImplAttribute(MethodImplOptions.InternalCall)]
-        extern private void DisposeNative();
+            // If we didn't find the exact device name in our pre-built bus list, bail out.
+            if( retVal == -1 )
+                throw new ArgumentException( );
```
Comment (ColinA-MSFT) [inline Framework/Core/Windows/Devices/Spi/SpiDevice.cs]: Braces here too.

File: DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp (hunk 2 above)
Comment (smaillet-ms) [inline DeviceCode/Targets/Native/STM32F4/DeviceCode/STM32F4_SPI/STM32F4_SPI_functions.cpp]: There's nothing unsafe about it, it's a style issue - overruled.

File: Framework/Core/Windows/Devices/Spi/SpiDevice.cs (hunk 4 above)
Comment (smaillet-ms) [inline Framework/Core/Windows/Devices/Spi/SpiDevice.cs]: Again, disagree.

Comment (ColinA-MSFT) [inline Framework/Core/Windows/Devices/Spi/SpiDevice.cs]: Please match the prevailing style in this file.

**Comment by ColinA-MSFT** [comment]
:+1:

File: CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiBusInfo.cpp (hunk 1 above)
Comment (smaillet-ms) [inline CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiBusInfo.cpp]: hmm... Why did we change that from what's generated?

Comment (ColinA-MSFT) [inline CLR/Libraries/Windows_Devices/windows_devices_native_Windows_Devices_Spi_SpiBusInfo.cpp]: This isn't a generated file.  Casing matches other core libraries.
